src/easier68k/core/enum/ea_mode.py

```diff
@@ -0,0 +1,145 @@
+"""
+Represents an effective addressing mode
+and methods associated with it
+"""
+from src.easier68k.core.util.parsing import parse_literal
+
+
+class EAMode:
+    # Enum Values
+    # Error value
+    ERR = -1
+
+    # Data register direct
+    DRD = 0
+
+    # Address register direct
+    ARD = 1
+
+    # Address register indirect
+    ARI = 2
+
+    # Address register indirect + post increment
+    ARIPI = 3
+
+    # Address register indirect + pre decrement
+    ARIPD = 4
+
+    # Immediate
+    IMM = 5
+
+    # Absolute long address
+    ALA = 6
+
+    # Absolute word address
+    AWA = 7
+
+    # Instance Values
+    # Which mode this represents
+    _mode = ERR
+
+    # Data value (could be register number, immediate data, or absolute address
+    _data = -1
+
+    def __init__(self, mode=ERR, data=-1):
+        self._mode = mode
+        self._data = data
+
+    def __str__(self):
+        return "Mode: {}, Data: {}".format(self._mode, self._data)
+
+    @staticmethod
+    def parse_ea(addr: str):
+        """
+        Parses an effective addressing mode (such as D0, (A1), #$01)
+
+        >>> EAMode.parse_ea('D')
+        Traceback (most recent call last):
+        ...
+        AssertionError
+
+        >>> str(EAMode.parse_ea('D3'))
+        'Mode: 0, Data: 3'
+
+        >>> str(EAMode.parse_ea('A6'))
+        'Mode: 1, Data: 6'
+
+        >>> str(EAMode.parse_ea('(A4)'))
+        'Mode: 2, Data: 4'
+
+        >>> str(EAMode.parse_ea('(A2)+'))
+        'Mode: 3, Data: 2'
+
+        >>> str(EAMode.parse_ea('(A2)-'))  # Invalid, can't do "post-decrement"
+        Traceback (most recent call last):
+        ...
+        AssertionError
+
+        >>> str(EAMode.parse_ea('($45).W'))
+        "Mode: 7, Data: bytearray(b'E')"
+
+        >>> str(EAMode.parse_ea('(%01010111).L'))
+        "Mode: 6, Data: bytearray(b'W')"
+
+        >>> str(EAMode.parse_ea('#$FF'))
+        "Mode: 5, Data: bytearray(b'\\\\xff')"
+
+        >>> str(EAMode.parse_ea('-(A2)'))
+        'Mode: 4, Data: 2'
+        """
+        assert len(addr) >= 2
```

Comment:
IIRC, instructions can be upper or lowercase, but I don't know if that is the same case for registers

Reply:
I might do some testing and find out if I need to watch cases on my assertions.

Reply:
Updated, can confirm: instructions and registers are all case insensitive. This makes it pretty easy to handle most things (except string literals, so we'll need to keep an eye out that we don't include those in any "uppercase" calls)